beeai/agents/tasks.py

```diff
@@ -69,7 +69,15 @@ async def commit_push_and_open_mr(
         files_to_commit = [files_to_commit]
     for path in itertools.chain(*(local_clone.glob(pat) for pat in files_to_commit)):
         await check_subprocess(["git", "add", str(path)], cwd=local_clone)
-    # TODO: check for empty commit (the command below will fail anyway, but we need to handle this somehow)
+    # Check if any files are staged before committing, if none, bail
+    exit_code, _, _ = await run_subprocess(
+        ["git", "diff", "--cached", "--quiet"],
+        cwd=local_clone,
+    )
+    # 1 = staged, 0 = none staged
+    if exit_code == 0:
+        logger.info("No files staged for commit, halting.")
+        return "No files staged for commit, halting."
```

Comment:
We should probably raise an exception here, this will not halt the process, it will let it continue with MR URL set to that text :slightly_smiling_face:

Reply:
Why would it set status to failed? There is no such logic in the code. It will succeed with `merge_request_url: No files staged for commit, halting.`.